src/main/java/seedu/address/model/team/UniqueTeamList.java

```diff
@@ -0,0 +1,150 @@
+        requireNonNull(toCheck);
+        return internalList.stream().anyMatch(toCheck::isSameTeam);
+    }
+
+    /**
+     * Adds a team to the list.
+     * The team must not already exist in the list.
+     */
+    public void add(Team toAdd) {
+        requireNonNull(toAdd);
+        if (contains(toAdd)) {
+            throw new DuplicateTeamException();
+        }
+        internalList.add(toAdd);
+    }
+
+    /**
+     * Replaces the team {@code target} in the list with {@code editedTeam}.
+     * {@code target} must exist in the list.
+     * The team identity of {@code editedTeam} must not be the same as another existing team in the list.
+     */
+    public void setTeam(Team target, Team editedTeam) {
+        requireAllNonNull(target, editedTeam);
+
+        int index = internalList.indexOf(target);
+        if (index == -1) {
+            throw new TeamNotFoundException();
+        }
+
+        if (!target.isSameTeam(editedTeam) && contains(editedTeam)) {
+            throw new DuplicateTeamException();
+        }
+
+        internalList.set(index, editedTeam);
+    }
+
+    /**
+     * Removes the equivalent team from the list.
+     * The team must exist in the list.
+     */
+    public void remove(Team toRemove) {
+        requireNonNull(toRemove);
+        if (!internalList.remove(toRemove)) {
+            throw new TeamNotFoundException();
+        }
+    }
+
+    public void setTeams(UniqueTeamList replacement) {
+        requireNonNull(replacement);
+        internalList.setAll(replacement.internalList);
+    }
+
+    /**
+     * Replaces the contents of this list with {@code teams}.
+     * {@code teams} must not contain duplicate teams.
+     */
+    public void setTeams(List<Team> teams) {
+        requireAllNonNull(teams);
+        if (!teamsAreUnique(teams)) {
+            throw new DuplicateTeamException();
+        }
+
+        internalList.setAll(teams);
+    }
+
+    /**
+     * Returns the backing list as an unmodifiable {@code ObservableList}.
+     */
+    public ObservableList<Team> asUnmodifiableObservableList() {
+        return internalUnmodifiableList;
+    }
+
+    @Override
+    public Iterator<Team> iterator() {
+        return internalList.iterator();
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        if (other == this) {
+            return true;
+        }
+
+        // instanceof handles nulls
+        if (!(other instanceof UniqueTeamList)) {
+            return false;
+        }
+
+        UniqueTeamList otherUniqueTeamList = (UniqueTeamList) other;
+        return internalList.equals(otherUniqueTeamList.internalList);
+    }
+
+    @Override
+    public int hashCode() {
+        return internalList.hashCode();
+    }
+
+    @Override
+    public String toString() {
+        return internalList.toString();
+    }
+
+    /**
+     * Returns true if {@code teams} contains only unique teams.
+     */
+    private boolean teamsAreUnique(List<Team> teams) {
```

Comment:
not sure if this is needed at all, if the set and edit functions already ensure uniqueneess

Reply:
This is called in `setTeams()`, which sets all members of a team at the same time, which is used for example when we load a saved team from storage. Therefore, I think we can still keep this method.